### PR TITLE
Feature/pid control

### PR DESCRIPTION
**Pull request by NerdboyQ**

### CODE REVIEW

File: arduino_pid_controller/code/btCommunication/BT_Communication_Standard.cpp [hunk 1]
```diff
@@ -13,16 +13,15 @@ FullBtMsg BTComm_Standard::Process_Request(byte pi_request[], byte length){
...
   byte upper_nibble = command_byte & 0b10000000;
   upper_nibble = upper_nibble >> 7;
   byte return_bit = 0;
   if (upper_nibble == 0){
-    return_bit = 0;
+    return_bit = false;
   }
   else{
-    return_bit = 1;
+    return_bit = true;
```
Comment (adonaygyb) [inline arduino_pid_controller/code/btCommunication/BT_Communication_Standard.cpp:24]: Love this. Boolean is much easier to understand for readers!!

File: arduino_pid_controller/code/btCommunication/PID_Controller.cpp [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  Serial.print("[P: "); 
+  Serial.print(P); 
+  Serial.print(", I: "); 
+  Serial.print(I); 
+  Serial.print(", D: "); 
+  Serial.print(D); 
+  Serial.print("];\t PID_out: "); 
+  Serial.println(PID_out);
+  Serial.print("servoPosition: ");
+  Serial.println(currentServoPosition);
```
Comment (adonaygyb) [inline arduino_pid_controller/code/btCommunication/PID_Controller.cpp:42]: Do you want to print this much?? Can this be the issue with the serial monitor?

File: arduino_pid_controller/code/btCommunication/PID_Controller.h [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    int currentServoPosition = 90; // sets the balance beam parallel to the surface
+    int targetServoPosition;
+
+    // Variables to use for calculating the min & max values
+    const int max_angle = 110; 
+    const int min_angle = 78; 
+    const int max_distance = 20; 
+    const int min_distance = 9; 
+    const int min_error = setpoint - max_distance;
+    const int max_error = setpoint - min_distance;
```
Comment (adonaygyb) [inline arduino_pid_controller/code/btCommunication/PID_Controller.h:41]: header file should just declare elements. Setting them to values should be handled in the constructer function.

File: arduino_pid_controller/code/btCommunication/PID_Controller.cpp (hunk 2 above)
Comment (NerdboyQ) [reply arduino_pid_controller/code/btCommunication/PID_Controller.cpp:42]: I can comment those lines out, there were being used for testing.  In the real practice, the serial monitor isn't used.

**Comment by adonaygyb** [comment]
where is the change from org to int in device.py? I don't see that here...

File: arduino_pid_controller/code/btCommunication/PID_Controller.cpp [hunk 4]
```diff
@@ -0,0 +1,50 @@
+#include "PID_Controller.h"
+
+PID_Controller::PID_Controller(){
+ // default code
+ previousTime = 0;
+ eventInterval = 250;
+ currentTime = millis();
+ currentServoPosition = 90;
+ setpoint = 16;
```
Comment (adonaygyb) [inline arduino_pid_controller/code/btCommunication/PID_Controller.cpp:9]: 👍

**Review by adonaygyb — APPROVED** [review]

**Pull request merged**